src/components/MapView/DateSelector/index.tsx

```diff
@@ -31,41 +36,109 @@ const Input = forwardRef(
   },
 );
 
-function DateSelector({ unavailableDates = [], classes }: DateSelectorProps) {
+function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
   const selectedDate = useSelector(selectCurrentDate);
+  const selectedMonth = selectedDate.getMonth();
+  const selectedYear = selectedDate.getFullYear();
   const dispatch = useDispatch();
 
-  function isUnavailableDate(date: Date) {
+  function isAvailableDate(date: Date) {
     return Boolean(
-      unavailableDates.find(dateIt => moment(date).isSame(dateIt, 'day')),
+      availableDates.find(dateIt => moment(date).isSame(dateIt, 'day')),
     );
   }
 
+  function isAvailableMonth(month: number) {
+    return Boolean(
+      findAvailableDayInMonth(new Date(selectedYear, month, 1), availableDates),
+    );
+  }
+
+  function updateMonth(month: number) {
+    // Fetching the first available day in the month
+    const date = findAvailableDayInMonth(
+      new Date(selectedYear, month, 1),
+      availableDates,
+    );
+    if (date) {
+      dispatch(updateDate(date));
+    }
+  }
+
+  function incrementYear() {
+    dispatch(updateDate(new Date(selectedYear + 1, selectedMonth, 1)));
+  }
+
+  function decrementYear() {
+    dispatch(updateDate(new Date(selectedYear - 1, selectedMonth, 1)));
+  }
+
   function updateStartDate(date: Date) {
-    if (!isUnavailableDate(date)) {
+    if (isAvailableDate(date)) {
```

Comment:
Can we use the UI element itself for this check? e.g. this function _only_ is callable (via `disabled` or something) when the date actually is available, rendering this check redundant?

If that's hard, it's not a big deal.